src/components/StakingVaults/AllEarnOpportunities.tsx

```diff
@@ -21,27 +26,60 @@ export const AllEarnOpportunities = () => {
   const history = useHistory()
   const location = useLocation()
   const foxyInvestorFeatureFlag = useAppSelector(state => selectFeatureFlag(state, 'FoxyInvestor'))
+  const cosmosInvestorFlag = useAppSelector(state => selectFeatureFlag(state, 'CosmosInvestor'))
   const {
     state: { isConnected },
     dispatch,
   } = useWallet()
   const sortedVaults = useSortedYearnVaults()
   const { opportunities } = useFoxyBalances()
+  const { activeStakingOpportunities, stakingOpportunities } = useCosmosStakingBalances({
+    assetId: 'cosmos:cosmoshub-4/slip44:118',
+  })
+
+  const { cosmosGetStarted, cosmosStaking } = useModal()
 
   const foxyRows = foxyInvestorFeatureFlag ? opportunities : []
+  const cosmosActiveStakingArray = cosmosInvestorFlag ? activeStakingOpportunities : []
+  const cosmosStakingArray = cosmosInvestorFlag ? stakingOpportunities : []
 
   const allRows = useNormalizeOpportunities({
     vaultArray: sortedVaults,
     foxyArray: foxyRows,
+    cosmosActiveStakingArray,
+    cosmosStakingArray,
   })
 
   const handleClick = useCallback(
     (opportunity: EarnOpportunityType) => {
-      const { type, provider, contractAddress, chain, tokenAddress, rewardAddress } = opportunity
+      const {
+        type,
+        provider,
+        contractAddress,
+        chain,
+        tokenAddress,
+        rewardAddress,
+        assetId,
+        cryptoAmount,
+      } = opportunity
       if (!isConnected) {
         dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
         return
       }
+
+      if (type === DefiType.TokenStaking && chain === ChainTypes.Cosmos) {
```

Comment:
Won't all Cosmos SDK opportunities be `TokenStaking` type?

Reply:
Hmm probably, I put it there as a security, should I remove the type check ?

Reply:
Yeah, unless there's a rationale to checking this, we shouldn't keep it here since for now, we will not have any other `DefiType` for Cosmos

Reply:
Fixed !

Reply:
Still there, reopening

Reply:
Oops yes there was 2 places with this, fixed !